openfisca_core/simulation_builder.py

```diff
@@ -557,25 +557,25 @@ def transform_to_strict_syntax(data):
     return data
 
 
-def iter_over_entity_members(entity_description, scenario_entity):
-    # One by one, yield individu_role, individy_legacy_role, individu_id
-    legacy_role_i = 0
-    for role in entity_description.roles:
+def iter_over_entity_members(entity, roles_json, instance_id):
```

Comment:
It feels very awkward to add instance_id as a parameter to this function, which does not need it other than to compose an error message.

In fact, I suspect this would be simpler, and this function wouldn't be needed at all, if (in `add_group_entity`) we iterated over the keys in the JSON rather than iterating over the roles in the entity.

Reply:
Agreed, this function is a remain from the scenarios and could be better integrated in the rest of the class. Trying to find something better.

Reply:
Done